sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/CacheTableExec.scala

```diff
@@ -86,12 +88,26 @@ case class CacheTableExec(
 case class CacheTableAsSelectExec(
     tempViewName: String,
     query: LogicalPlan,
+    originalText: Option[String],
     override val isLazy: Boolean,
     override val options: Map[String, String]) extends BaseCacheTableExec {
   override lazy val relationName: String = tempViewName
 
   override lazy val planToCache: LogicalPlan = {
-    Dataset.ofRows(sparkSession, query).createTempView(tempViewName)
+    val viewIdent = sparkSession.sessionState.sqlParser.parseTableIdentifier(tempViewName)
```

Comment:
The `tempViewName` is already a single-part name produced by the parser,  and we shouldn't reparse it. We can just pass in `TableIdentifier(tempViewName)`.